go/interpreter/lfvm/memory_test.go

```diff
@@ -41,9 +44,475 @@ func TestExpansionCosts(t *testing.T) {
 	for _, test := range tests {
 
 		m := NewMemory()
-		cost := m.ExpansionCosts(test.size)
+		cost := m.getExpansionCosts(test.size)
 		if cost != test.cost {
 			t.Errorf("ExpansionCosts(%d) = %d, want %d", test.size, cost, test.cost)
 		}
 	}
 }
+
+func TestMemory_GetWord_CopiesData(t *testing.T) {
+
+	valueSmall := uint256.NewInt(0x1223457890abcdef)
+	valueMiddle := uint256.NewInt(0).Lsh(valueSmall, 64)
+	valueBig := uint256.NewInt(0).Lsh(valueSmall, 256-16)
+	memorySize := uint64(32)
+
+	tests := map[string]struct {
+		offset       uint64
+		expectedData *uint256.Int
+	}{
+		"regular": {
+			offset:       0,
+			expectedData: valueSmall,
+		},
+		"small offset": {
+			offset:       memorySize / 4,
+			expectedData: valueMiddle,
+		},
+		"big offset crops value": {
+			offset:       memorySize - 2,
+			expectedData: valueBig,
+		},
+		"big offset returns zero": {
+			offset:       memorySize,
+			expectedData: uint256.NewInt(0),
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			ctxt := getEmptyContext()
+			m := NewMemory()
+			target := uint256.NewInt(1)
+			m.store = make([]byte, memorySize)
+			copy(m.store[24:], valueSmall.Bytes())
+			ctxt.gas = 100
+
+			err := m.readWord(test.offset, target, &ctxt)
+			if err != nil {
+				t.Fatalf("unexpected error, want: %v, got: %v", nil, err)
+			}
+			if target.Cmp(test.expectedData) != 0 {
+				t.Errorf("unexpected target value, want: %x, got: %x", *test.expectedData, *target)
+			}
+		})
+	}
+}
```

Comment:
I have been reading this test so many times in the last week ... an every time I get a full mental overload. It takes me forever to get what it tries to do and to I never get full confidence that it does what I think it is doing. One of the problems is the naming of stuff, another that one needs to connect multiple different definitions from within this single function to get what is tested. Yet another, that the test covers so many aspects that are just implicitly taken for granted (e.g. byte-granularity addressing, big endian encoding of integer values in the memory, ...). To cover all of this, this test should be named `TestMemory_GetWord_RetrievesDataAtGivenByteOffsetInProperNumericalInterpretationAndAddsZeroPadding`.

The word-addressing and zero padding could for instance be tested like this:
```
func TestMemory_GetWord_ReadsDataWithByteAddressingAndZeroPadding(t *testing.T) {
	pattern := make([]byte, 32)
	for i := 0; i < 32; i++ {
		pattern[i] = byte(i + 1)
	}

	for i := 0; i <= 32; i++ {
		ctxt := getEmptyContext()
		ctxt.gas = 100

		m := NewMemory()
		m.expandMemory(32)
		copy(m.store, pattern)

		var target uint256.Int
		err := m.getWord(uint64(i), &target, &ctxt)
		if err != nil {
			t.Fatalf("unexpected error, want: %v, got: %v", nil, err)
		}

		// Check that the memory pattern is shifted by i bytes to the left.
		want := [32]byte{}
		copy(want[:], pattern[i:])
		if got := target.Bytes32(); want != got {
			t.Errorf("unexpected value, want: %x, got: %x", want, got)
		}
	}
}
```
This test covers the same property as the test you provided, covers it for all offsets between 0-32, does not require anyone to think about the byte- and word-order `uint256` is using and the effect the left-shift may have on it, and is significantly shorter.

I think you need to focus more on the quality of the tests and their comprehensibility.